tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,685 @@
+            X509v3 Basic Constraints: 
+                CA:FALSE
+    Signature Algorithm: sha256WithRSAEncryption
+         75:41:37:[...]
+```
+
+No we are all set to install the certificates.
+
+### Certificate Chains
+
+For the sake of completeness it should be mentioned here that out in the wild server certificates are usually signed by subordinate CAs. So the client has to follow the certificate chain all the way down to a CA it trusts. Servers can even present server certificates that are combined with some intermediate CA certificates to fill the gaps. Even then the fact still remains, that client must know and trust the last CA in the chain.
+
+We keep things simple here though, since this is just a tutorial meant to convey the general idea of server certificate validation.
+
+## Setting up MySQL
+
+1. Install MySQL client and server
+   ```shell
+   sudo apt install mysql-client mysql-server
+   ```
+2. Log into MySQL as MySQL root user (on Ubuntu now requires `sudo`)
+   ```shell
+   sudo mysql -u root
+   ```
+3. Create a database `shapes` and a user `tutorial_user` for our import experiment
+   ```sql
+   create database shapes;
+   create user 'tutorial_user' identified by 'tutorial';
+   grant all on shapes.* to tutorial_user;
+   exit
+   ```
+4. Login as a `tutorial_user`
+   ```sql
+   mysql -u tutorial_user -p
+   ```
+5. Populate the database
+   ```sql
+   create table shapes.shapes (name VARCHAR(40), corners int);
+   insert into shapes.shapes values ('point', 1), ('line', 2), ('triangle', 3), ('rectangle', 4);
+   exit
+   ```
+
+As you can tell we won't win any price for most original database usage here, but that is not the point of the tutorial.
+
+### Installing Certificates in MySQL
+
+By default, the MySQL server on Ubuntu comes with a self-signed certificate. Our next job will be to use the certificates we previously created.
+
+1. Install the CA certificate centrally:
+   ```shell
+   sudo cp ca.crt /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   sudo update-ca-certificates
+   ```
+2. Verify that it was installed:
+   ```shell
+   ls -l /etc/ssl/certs/exasol*
+   ```
+   Must output something like:
+   ```
+   lrwxrwxrwx 1 root root 55 Jan 12 11:43 /etc/ssl/certs/exasol_tutorial_ca.pem -> /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   ```
+   Note how the symbolic link changes the file suffice from `.crt` to `.pem`
+3. Install the server certificate and key
+   ```shell
+   sudo mkdir -p /etc/mysql/certs
+   sudo cp server.crt /etc/mysql/certs/
+   sudo cp server.key /etc/mysql/certs/
+   sudo chown mysql:mysql /etc/mysql/certs/server.key
+   ```
+
+Note that we need `sudo` privileges here, since we need to install the certificates in a central locations that belong to the `root` user.
+
+Additionally, we give the private key to the MySQL user. We don't want anyone else to be able to read it, so weakening permissions on that file is not an option.
```

Comment:
Should be mentioned that this user is automatically created upon install of MySQL.